### PR TITLE
Move callbacks to source code

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Note: `get_callbacks` and `turb_conv_affect_filter!` are still in examples (in the new file `get_callbacks.jl`), because they use `TurbulenceConvectionUtils` in examples.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/callbacks.jl [hunk 1]
```diff
@@ -131,7 +130,7 @@ function dss_callback(integrator)
     Y = integrator.u
     ghost_buffer = integrator.p.ghost_buffer
     if !ghost_buffer.skip_dss
-        @nvtx "dss callback" color = colorant"yellow" begin
+        NVTX.@range "dss callback" color = colorant"yellow" begin
```
Comment (jiahe23) [inline src/callbacks.jl:133]: Why is this change needed?

**Comment by jiahe23** [comment]
LGTM

**Review by jiahe23 — APPROVED** [review]

File: src/callbacks.jl (hunk 1 above)
Comment (szy21) [reply src/callbacks.jl:133]: The @nvtx macro is defined in example (https://github.com/CliMA/ClimaAtmos.jl/blob/main/examples/hybrid/nvtx.jl) and cannot be used in src. But it is essentially the same as NVTX.@range.

**Comment by szy21** [comment]
bors try
